django/contrib/auth/__init__.py

```diff
@@ -108,7 +109,14 @@ def login(request, user, backend=None):
         request.session.cycle_key()
 
     try:
+
         backend = backend or user.backend
+        # backend should be a dotted import path string
```

Comment:
If the comment is useful, maybe it would be better to put it in the error message.

Reply:
I agree—this is more specific than just saying that it has to be a string.